### PR TITLE
[WIP]:Pseudo-Transient method

### PR DESCRIPTION
**Pull request by yonatanwesen**
This PR implements pseudo transient method using the adaptive step size given https://www.semanticscholar.org/paper/Explicit-pseudo-transient-continuation-Kelley-Liao/57061a163326620d425435d98e51fbe0aa14f1cc?p2df

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
That URL doesn't work. Which paper?

File: src/NonlinearSolve.jl [hunk 1]
```diff
@@ -37,6 +37,7 @@ include("utils.jl")
 include("raphson.jl")
 include("trustRegion.jl")
 include("levenberg.jl")
+include("pseudoTransient.jl")
```
Comment (ChrisRackauckas) [inline src/NonlinearSolve.jl:40]: please follow standard naming conventions. `psudotransient.jl` for the page should be fine.

File: src/jacobian.jl [hunk 2]
```diff
@@ -105,16 +105,25 @@ function build_jac_and_jac_config(alg, f::F1, uf::F2, du1, u, tmp, du2) where {F
...
 
     du1 = zero(u)
     du2 = zero(u)
     tmp = zero(u)
     J, jac_config = build_jac_and_jac_config(alg, f, uf, du1, u, tmp, du2)
+    
+    if alg isa PseudoTransient
+        alpha = convert(eltype(u),alg.alpha_initial)
+        J_new = J - (1/alpha) * I
+        linprob = LinearProblem(J_new, _vec(zero(u)); u0 = _vec(zero(u)))
```
Comment (ChrisRackauckas) [inline src/jacobian.jl:120]: What difference does this make?

Comment (avik-pal) [reply src/jacobian.jl:120]: If `J` is a VecJac operator, then the addition constructs an `AddedOperator`. needed for Krylov Subspace compatibility.

Comment (avik-pal) [reply src/jacobian.jl:120]: Without this the types wont match in `dolinsolve`

**Comment by avik-pal** [comment]
Link to paper https://www.semanticscholar.org/paper/Explicit-pseudo-transient-continuation-Kelley-Liao/57061a163326620d425435d98e51fbe0aa14f1cc?p2df

File: src/pseudotransient.jl [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    return nothing
+end
+
+function perform_step!(cache::PseudoTransientCache{false})
+    @unpack u, fu1, f, p, alg, linsolve, alpha = cache
+
+    cache.J = jacobian!!(cache.J, cache)
+    # u = u - J \ fu
+    if linsolve === nothing
+        cache.du = fu1 / cache.J
```
Comment (avik-pal) [inline src/pseudotransient.jl:92]: This is incorrect. Needs the $\frac{1}{\alpha}$ term

**Comment by avik-pal** [comment]
Open a new PR for this. Also, set your git config to rebase and not merge on pull. Otherwise, these disastrous merges happen.

**Pull request closed without merging**